containers/src/Data/Set/Internal.hs

```diff
@@ -1308,6 +1308,24 @@ splitMember x (Bin _ y l r)
 {-# INLINABLE splitMember #-}
 #endif
 
+-- Same as 'splitMember' but skips re-balancing by using 'bin' instead of 'link'.
+-- Attempting to build new trees out of these will error when re-balancing but
+-- this can improve performance when the resulting trees are disposable.
+splitMemberUnbalanced :: Ord a => a -> Set a -> (Set a,Bool,Set a)
```

Comment:
Tried to factor out `bin`/`link` to avoid code duplication but that made the involved functions 2-3x slower.